script.recap/resources/lib/core.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+import xbmc
+import xbmcaddon
+import xbmcgui
+
+# Import the common settings
+from settings import log
+from settings import Settings
+from recap import Recap
+from viewer import Viewer
+
+ADDON = xbmcaddon.Addon(id='script.recap')
+
+
+#########################
+# Main
+#########################
+def runRecap(previous=False):
+    log("runRecap: Started")
+
+    # First check to see if we have a TV Show already selected
+    showName = None
+    if xbmc.getInfoLabel("ListItem.dbtype") in ['tvshow', 'season', 'episode']:
```

Comment:
This way wont work when triggered from any non-media window (home screen for example) via context menu. You need to check sys.listitem in that case, otherwise it will be buggy in krypton.